### PR TITLE
[libwallet] Set confirmations required via libwallet and config

### PR DESCRIPTION
**Pull request by delta1**
## Description

Adds a config option, transaction service methods, and libwallet FFI functions to set the number of block confirmations required for the wallet transaction service.

## Motivation and Context

Allows number of confirmations required to be set from config file or programmatically. 

## How Has This Been Tested?

Tests provided (some previously broken tests are being fixed in #2613) 

## Types of changes

* [ ] Bug fix (non-breaking change which fixes an issue)
* [x] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [x] Feature refactor (No new feature or functional changes, but performance or technical debt improvements)
* [x] New Tests
* [ ] Documentation

## Checklist:

* [x] I'm merging against the `development` branch.
* [x] I ran `cargo-fmt --all` before pushing.
* [ ] I ran `cargo test` successfully before submitting my PR.
* [x] I have squashed my commits into a single commit.
* [x] My change requires a change to the documentation.
* [x] I have updated the documentation accordingly.
* [x] I have added tests to cover my changes.

### CODE REVIEW

File: base_layer/wallet/src/transaction_service/config.rs [hunk 1]
```diff
@@ -24,6 +24,7 @@ use log::*;
 use std::{fmt, time::Duration};
 
 const LOG_TARGET: &str = "wallet::transaction_service::config";
+pub const NUM_CONFIRMATIONS_REQUIRED_DEFAULT: u64 = 3;
```
Comment (philipr-za) [inline base_layer/wallet/src/transaction_service/config.rs]: Style Nit but is putting this in a const really neater than all of the info being in the Default trait implementation?

File: base_layer/wallet_ffi/src/lib.rs [hunk 2]
```diff
@@ -5862,6 +5940,21 @@ mod test {
                 .runtime
                 .block_on((*bob_wallet).wallet.db.set_chain_meta(meta_data.clone()))
                 .unwrap();
+
+            // number of confirmations
+            let num_confirmations_required = wallet_get_num_confirmations_required(alice_wallet, error_ptr);
+            assert_eq!(num_confirmations_required, NUM_CONFIRMATIONS_REQUIRED_DEFAULT);
```
Comment (philipr-za) [inline base_layer/wallet_ffi/src/lib.rs]: So here I would use `TransactionServiceConfig::default().num_confirmations_required`

**Review by philipr-za — DISMISSED** [review]
Looks good, one style nit

File: base_layer/wallet_ffi/src/lib.rs (hunk 2 above)
Comment (delta1) [reply base_layer/wallet_ffi/src/lib.rs]: I agree that is neater 👌 will do

File: base_layer/wallet/src/transaction_service/config.rs (hunk 1 above)
Comment (delta1) [reply base_layer/wallet/src/transaction_service/config.rs]: I never thought of that, good idea. Thanks @philipr-za

**Review by philipr-za — DISMISSED** [review]

**Pull request merged**